packages/origin-247-certificate/src/evented-certificate/events/Certificate.events.ts

```diff
@@ -0,0 +1,74 @@
+import { IEvent } from '@nestjs/cqrs';
+import { IIssueCommand, ITransferCommand, IClaimCommand } from '../../types';
+namespace Certificate {}
+export enum CertificateEventType {
+    Issued = 'Issued',
+    Transferred = 'Transferred',
+    Claimed = 'Claimed',
+    IssuancePersisted = 'IssuancePersisted',
+    TransferPersisted = 'TransferPersisted',
+    ClaimPersisted = 'ClaimPersisted'
+}
+
+export class CertificateIssuedEvent implements IEvent {
+    public readonly type = CertificateEventType.Issued;
+
+    constructor(
+        public internalCertificateId: number,
+        public version: number,
```

Comment:
So you don't implement version like that.
It's not up to event user to determine version, it's up to event to do it.

So basically, this version should be hardcoded on event just like `type`. Then, the version will determine payload properly. Because if `payload` interface changes, then we juts bump the version up on the event, and there is no problem - all new events saved in database will get new version automatically.